src/reader/geometry.rs

```diff
@@ -12,44 +12,37 @@ use geo_traits::{
     Dimensions, GeometryTrait, UnimplementedLine, UnimplementedRect, UnimplementedTriangle,
 };
 
+/// Parse a WKB byte slice into a geometry.
+///
+/// An opaque object that implements [`GeometryTrait`]. Use methods provided by [`geo_traits`] to
+/// access the underlying data.
+///
+/// The contained [dimension][geo_traits::Dimensions] will never be `Unknown`.
 #[derive(Debug, Clone)]
-pub enum Wkb<'a> {
-    Point(Point<'a>),
-    LineString(LineString<'a>),
-    Polygon(Polygon<'a>),
-    MultiPoint(MultiPoint<'a>),
-    MultiLineString(MultiLineString<'a>),
-    MultiPolygon(MultiPolygon<'a>),
-    GeometryCollection(GeometryCollection<'a>),
-}
+pub struct Wkb<'a>(WkbInner<'a>);
```

Comment:
I made a wrapper struct so that not even the `Wkb` variants are public, though I can roll it back if desired.